blockdag/blocknode.go

```diff
@@ -198,6 +204,22 @@ func (node *blockNode) RelativeAncestor(distance uint64) *blockNode {
 	return node.SelectedAncestor(node.chainHeight - distance)
 }
 
+// isInPast returns true iff this node can be reached from the other in the DAG.
+// The complexity of this method is O(log(|node.futureBlocks|))
+func (node *blockNode) isInPast(other *blockNode) bool {
+	// First, check if this node is a reachability tree ancestor of the
+	// other node
+	thisTreeInterval := node.reachabilityTreeNode.interval
+	otherTreeInterval := other.reachabilityTreeNode.interval
+	if thisTreeInterval.isAncestorOf(&otherTreeInterval) {
```

Comment:
This is a bit weird that we have node.isInPase(otherNode), but than interval.isAncerstorOf(otherInterval).
Please rename one of them.